File: Tests/SkeletonTypeScriptParserTests/Fixtures/Sample.ts
interface Printable {
    name: string;
    print(): void;
}

interface Serializable {
    serialize(): string;
}

class Animal implements Printable {
    name: string;
    age: number;

    constructor(name: string, age: number) {
        this.name = name;
        this.age = age;
    }

    print(): void {
        console.log(this.name);
    }

    greet(greeting: string): string {
        return `${greeting}, ${this.name}`;
    }
}

class Dog extends Animal implements Serializable {
    breed: string;

    constructor(name: string, age: number, breed: string) {
        super(name, age);
        this.breed = breed;
    }

    serialize(): string {
        return JSON.stringify(this);
    }

    fetch(item: string): boolean {
        return true;
    }
}

enum Direction {
    Up = "UP",
    Down = "DOWN",
    Left = "LEFT",
    Right = "RIGHT",
}

type Point = {
    x: number;
    y: number;
};

File: Tests/SkeletonGoParserTests/Fixtures/sample.go
package sample

type Stringer interface {
	String() string
}

type Animal struct {
	Name string
	Age  int
}

func (a *Animal) String() string {
	return a.Name
}

func (a *Animal) Greet(greeting string) string {
	return greeting + " " + a.Name
}

type Dog struct {
	Animal
	Breed string
}

func (d *Dog) Fetch(item string) bool {
	return true
}

func NewAnimal(name string, age int) *Animal {
	return &Animal{Name: name, Age: age}
}
